File: Modules/Project/FiyonaMistry/Naukri.com/FrontEnd/src/app/shared/Models/Naukri/JobSeeker.ts
import { IJobSeeker } from '../INaukri/IJobSeeker';

export class JobSeeker implements IJobSeeker{
    jobSeekerId? : number;
    jobSeekerFirstName : string;
    jobSeekerMiddleName : string;
    jobSeekerLastName : string;
    jobSeekerEmail : string;
    jobSeekerPassword : string;
    jobSeekerPhoneNumber : number;
    jobSeekerType : string;
    jobSeekerResume : string;
    jobSeekerSkills : string;
    jobSeekerDob : Date;
    jobSeekerGender : string;
    jobSeekerAddress : string;
    jobSeekerProjects? : string;
    jobSeekerAreaPinCode : number;
    id : number;
    userId : {}[];
    jobSeekerEducations : {
        JobSeekerEducationId : number;
        jobSeekerField : string;
        jobSeekerCollege : string;
        jobSeekerYearOfCompletion : number;
    }[];
    jobSeekerJobHistories? : {
        JobSeekerDesignation : string;
        JobSeekerCompany : string;
        JobSeekerAnnualSalary : number;
        JobSeekerWorkingFrom : Date;
        JobSeekerWorkingTo : Date;
        JobSeekerCurrentCity : string;
        IsCurrent? : number;
    }[];

    constructor(jobSeekerId : number, fname : string, mname : string, lname : string, email : string, password : string, phonenumber : number, type : string, resume : string, skills : string, dob : Date, gender : string, jobSeekerProjects : string, address : string, areapincode : number, id : number, userId : string, jobSeekerEducations : Array<any>, jobSeekerJobHistories : Array<any>){
        this.jobSeekerId = jobSeekerId;
        this.jobSeekerFirstName = fname;
        this.jobSeekerMiddleName = mname;
        this.jobSeekerLastName = lname;
        this.jobSeekerEmail = email;
        this.jobSeekerPassword = password;
        this.jobSeekerPhoneNumber = phonenumber;
        this.jobSeekerType = type;
        this.jobSeekerResume = resume;
        this.jobSeekerSkills = skills;
        this.jobSeekerDob = dob;
        this.jobSeekerGender = gender;
        this.jobSeekerAddress = address;
        this.jobSeekerProjects = jobSeekerProjects;
        this.id = id;
        this.userId = [userId];
        this.jobSeekerAreaPinCode = areapincode;
        this.jobSeekerEducations = jobSeekerEducations;
        this.jobSeekerJobHistories = jobSeekerJobHistories;
    }
}

File: Modules/Project/FiyonaMistry/Naukri.com/Web API/NaukriProject/NaukriProject/Models/JobSeeker.cs
using NaukriProject.Authentication;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

#nullable disable

namespace NaukriProject.Models
{
    public partial class JobSeeker
    {
        public JobSeeker()
        {
            JobSeekerEducations = new HashSet<JobSeekerEducation>();
            JobSeekerJobHistories = new HashSet<JobSeekerJobHistory>();
        }

        [Key]
        public int JobSeekerId { get; set; }
        public string JobSeekerFirstName { get; set; }
        public string JobSeekerMiddleName { get; set; }
        public string JobSeekerLastName { get; set; }
        public long JobSeekerPhoneNumber { get; set; }
        public string JobSeekerType { get; set; }
        public string JobSeekerEmail { get; set; }
        public string JobSeekerResume { get; set; }
        public string JobSeekerSkills { get; set; }
        public string JobSeekerProjects { get; set; }
        public DateTime? JobSeekerDob { get; set; }
        public string JobSeekerGender { get; set; }
        public string JobSeekerAddress { get; set; }
        public int? JobSeekerAreaPinCode { get; set; }
        public string JobSeekerPassword { get; set; }

        public string Id { get; set; }
        public ApplicationUser UserId { get; set; }

        //[System.Text.Json.Serialization.JsonIgnore]
        public virtual ICollection<JobSeekerEducation> JobSeekerEducations { get; set; }

        //[System.Text.Json.Serialization.JsonIgnore]
        public virtual ICollection<JobSeekerJobHistory> JobSeekerJobHistories { get; set; }
    }
}
